scripts/helpers/poll_quick_ci.py

```diff
@@ -0,0 +1,40 @@
+import argparse
+import subprocess
+import time
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("pr", help="Pull request number")
+    parser.add_argument("check", help="Pull request check to poll for")
+    args = parser.parse_args()
+
+    print(f"Gathering info on the {args.check} check being run for the current pull request.")
+    polling = True
+    poll_count = 0
+    poll_max = 300
+    while (polling):
+        for line in subprocess.run(f"gh pr checks -R project-chip/connectedhomeip {args.pr}", stdout=subprocess.PIPE, shell=True).stdout.decode("utf-8").splitlines():
+            if args.check in line:
+                print(line)
+                if "pending" in line:
+                    if poll_count == poll_max:
+                        polling = False
+                    else:
+                        poll_count += 1
+                        time.sleep(1)
+                    break
+                elif "pass" in line:
+                    print(f"Fast CI {args.check} has passed; if all fast CI passes, builds and tests may commence.")
+                    exit(0)
+                elif "fail" in line:
+                    print(f"Fast CI {args.check} has failed; please resolve this issue before running builds and tests.")
+                    exit(1)
+
+    print(
+        f"Polling for completion of fast CI {args.check} failed. Please ensure the name of the check was entered correctly and verify that it should take less than {poll_max} seconds to run.")
```

Comment:
Does this imply that if the fast jobs don't complete within 300s that this will fail and we'll cancel all the remaining CI jobs?

I think in this case maybe we should just let the complete? Thoughts @andy31415 ?

Reply:
Changed the timeout to 1 hour - if it takes longer, that implies the job is hanging.